SpatialGDK/Source/SpatialGDKTests/SpatialGDKEditor/SpatialGDKEditorSchemaGenerator/SpatialGDKEditorSchemaGeneratorTest.cpp

```diff
@@ -1051,4 +1051,155 @@ SCHEMA_GENERATOR_TEST(GIVEN_no_schema_exists_WHEN_generating_schema_for_rpc_endp
+
+	FString SchemaFolder = FPaths::Combine(SchemaOutputFolder, TEXT("schema"));
+	FString UnrealSchemaFolder = FPaths::Combine(SchemaFolder, TEXT("unreal"));
+	FString SchemaGenerationFolder = FPaths::Combine(UnrealSchemaFolder, TEXT("generated"));
+
+	// Generate data for well known classes.
+	SpatialGDKEditor::Schema::SpatialGDKGenerateSchemaForClasses(Classes, SchemaGenerationFolder);
+	USchemaDatabase* SchemaDatabase = SpatialGDKEditor::Schema::InitialiseSchemaDatabase(DatabaseOutputFile);
+	SpatialGDKEditor::Schema::WriteComponentSetFiles(SchemaDatabase, SchemaGenerationFolder);
+
+	FString SchemaBuildFolder = FPaths::Combine(SchemaOutputFolder, TEXT("Build"));
+
+	// Add the files necessary to run the schema compiler
+	FString GDKSchemaCopyDir = FPaths::Combine(UnrealSchemaFolder, TEXT("gdk"));
+	FString CoreSDKSchemaCopyDir = FPaths::Combine(SchemaBuildFolder, TEXT("dependencies/schema/standard_library"));
+	SpatialGDKEditor::Schema::CopyWellKnownSchemaFiles(GDKSchemaCopyDir, CoreSDKSchemaCopyDir);
+	SpatialGDKEditor::Schema::GenerateSchemaForRPCEndpoints(SchemaGenerationFolder);
+	SpatialGDKEditor::Schema::GenerateSchemaForNCDs(SchemaGenerationFolder);
+
+	// Run the schema compiler
+	FString SchemaJsonPath;
+
+	TestTrue("Schema compiler run successful",
+			 SpatialGDKEditor::Schema::RunSchemaCompiler(SchemaJsonPath, SchemaFolder, SchemaBuildFolder));
+
+	TestTrue("Schema bundle file successfully read",
+			 SpatialGDKEditor::Schema::ExtractComponentSetsFromSchemaJson(SchemaJsonPath, SchemaDatabase->ComponentSetIdToComponentIds));
+
+	TestTrue("Expected number of component set", SchemaDatabase->ComponentSetIdToComponentIds.Num() == 7);
+
+	TestTrue("Found spatial well known components", SchemaDatabase->ComponentSetIdToComponentIds.Contains(50));
+	if (SchemaDatabase->ComponentSetIdToComponentIds.Contains(50))
+	{
+		TestTrue("Spatial well know component is not empty", SchemaDatabase->ComponentSetIdToComponentIds[50].ComponentIDs.Num() > 0);
+	}
+
+	{
+		// Check the resulting schema contains the expected Sets.
+
+		FComponentIDs* ServerComponents = SchemaDatabase->ComponentSetIdToComponentIds.Find(SpatialConstants::SERVER_AUTH_COMPONENT_SET_ID);
+		TestTrue("Found entry for server authority", ServerComponents != nullptr);
+		if (ServerComponents == nullptr)
+		{
+			return false;
+		}
+		TestTrue("Set is not empty", ServerComponents->ComponentIDs.Num() > 0);
+		for (auto ComponentId : SpatialConstants::ServerAuthorityWellKnownComponents)
+		{
+			FString DebugString = FString::Printf(TEXT("Found well known component %s"), *ComponentId.Value);
+			TestTrue(*DebugString, ServerComponents->ComponentIDs.Find(ComponentId.Key) != INDEX_NONE);
+		}
+
+		uint32 ServerAuthSets[3] = { SpatialConstants::DATA_COMPONENT_SET_ID, SpatialConstants::OWNER_ONLY_COMPONENT_SET_ID,
+									 SpatialConstants::HANDOVER_COMPONENT_SET_ID };
+
+		for (uint32 DataType = SCHEMA_Data; DataType < SCHEMA_Count; ++DataType)
```

Comment:
This was confusing to read at first- as the first DataType is named Data, it wasn't clear where we were checking handover or owner only. Could we rename DataType to ComponentType or similar to avoid this?